.github/workflows/tests.yml

```diff
@@ -234,6 +234,13 @@ jobs:
             exit 1
           fi
 
+      # XXX: `make static-analysis` also run golangci-lint but this one provides
+      #      useful feedback in the PR through github-code-scanning bot
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@4afd733a84b1f43292c63897423277bb7f4313a9 # v8
+        with:
+          args: --timeout=5m --new --new-from-rev=${{ github.event.pull_request.base.sha }} --whole-files
```

Comment:
Does this work with "push" action for when its merged? 
I remember it took a long time to figure out something that worked for both.